### PR TITLE
[doc] updated get_started index page

### PR DESCRIPTION
**Pull request by mli**
preview at http://ec2-52-88-111-46.us-west-2.compute.amazonaws.com/get_started/index.html

TODO: scala/julia/r/perl examples are not completed

### CODE REVIEW

File: docs/get_started/index.md [hunk 1]
```diff
@@ -174,45 +90,149 @@ AWS images with MXNet installed:
...
+>>> c[0].asnumpy()  # the list of outputs
+array([[ 3.,  3.,  3.],
+       [ 3.,  3.,  3.]], dtype=float32)
+```
+
+Run the above codes in GPU in straightforward:
+
+```python
+>>> a = mx.nd.ones((2, 3), mx.gpu())  # create a on GPU 0, then the result a*2+1 will sit on GPU 0 as well
+>>> c = b.eval(a=a, mx.gpu()), ctx=mx.gpu())  # feed a as the input to eval b, the result c will be also on GPU 0
```
Comment (piiswrong) [inline docs/get_started/index.md]: b.eval(a=a, mx.gpu()), ctx=mx.gpu()) -> b.eval(a=a, ctx=mx.gpu())

**Comment by piiswrong** [comment]
When a installation method is not available there should be a warning message.
Currently nothing shows up when choosing R->Prebuilt Binaries

**Comment by mli** [comment]
fixed. thanks

File: docs/get_started/index.md [hunk 2]
```diff
@@ -11,141 +11,55 @@ imperative operations on the fly.  A graph optimization layer on top of that
...
-[1,]    2    2    2
-[2,]    2    2    2
-```
-
-</div> <!-- r-demo -->
-<div role="tabpanel" class="tab-pane" id="julia-demo">
-
-```julia
-julia> using MXNet
+Please choice the programming language for the rest of this document.
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/index.md]: may be 'choose' instead of 'choice'?

File: docs/get_started/index.md [hunk 3]
```diff
@@ -174,45 +90,149 @@ AWS images with MXNet installed:
...
 Refer to [docker/](../../docker/) for more details.
 
 </div> <!-- docker -->
 
 <div class="build-from-source">
 
-Refer to [setup](./setup.md) for details on building MXNet from source codes for
-various systems.
+Refer to the [building from source document](./build_from_source.md) for details
+on building MXNet from source codes for various systems.
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/index.md]: various platforms?

File: docs/get_started/index.md [hunk 4]
```diff
@@ -11,141 +11,70 @@ imperative operations on the fly.  A graph optimization layer on top of that
 makes symbolic execution fast and memory efficient. The MXNet library is
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/index.md]: 1. "Test MXNet installation" and "Common Installation Issue or Troubleshooting" are 2 sections that will be useful. Current "overview" section is serving purpose of "Test MXNet Installation".
2. In place of "Overview" we should probably have more better example as this is a getting started page.
3. Should we move the installation to separate page and link it from getting started page?
4. Programming languages in the middle of the page is somewhat not very good looking.

File: docs/get_started/index.md [hunk 5]
```diff
@@ -174,45 +105,149 @@ AWS images with MXNet installed:
...
+
+julia> Array{Float32}(a * 2 + 1)
+2×3 Array{Float32,2}:
+ 3.0  3.0  3.0
+ 3.0  3.0  3.0
+```
+
+```perl
+pdl> use AI::MXNet qw(mx)
+pdl> $a = mx->nd->ones([2, 3], ctx => mx->gpu())
```
Comment (sandeep-krishnamurthy) [inline docs/get_started/index.md]: May be we should have CPU context?

File: docs/get_started/index.md (hunk 4 above)
Comment (mli) [inline docs/get_started/index.md]: 1. overview is not severing as testing installation. it gives an quick overview of mxnet's interface. we can provide some test scripts for people to test. or simply make test
2. @zackchase is working on it
3. noop. @zackchase suggested to move the installation into the mxnet.io frontpage to reduce one mouse click
4. placing on the left is also ugly. i don't have a good solution yet. but i probably left it to our future web designers.

File: docs/get_started/index.md (hunk 5 above)
Comment (mli) [inline docs/get_started/index.md]: i don't have perl installed, so i don't want to make change unless i can test it.

File: docs/get_started/index.md (hunk 3 above)
Comment (mli) [inline docs/get_started/index.md]: fixed

**Comment by domdivakaruni** [comment]
When you go to How To and look for installation, it leads you to the instructions to build from source. Can we take folks to a consolidated view like http://mxnet.io/get_started/index.html#setup-mxnet instead?

**Pull request merged**